mhs/common/mhs_common/workflow/synchronous.py

```diff
@@ -1,17 +1,116 @@
-from typing import Tuple
-
+from typing import Tuple, Optional
+import utilities.integration_adaptors_logger as log
+from tornado import httpclient
+from utilities import timing
+from mhs_common import workflow
+from mhs_common.errors.soap_handler import handle_soap_error
+from mhs_common.messages import soap_envelope
+from mhs_common.state import persistence_adaptor as pa
+from mhs_common.state import work_description as wd
+from mhs_common.transmission import transmission_adaptor
 from mhs_common.workflow import common_synchronous
-import mhs_common.state.work_description as wd
+from mhs_common.routing import routing_reliability
+
+logger = log.IntegrationAdaptorsLogger('SYNC_WORKFLOW')
 
 
 class SynchronousWorkflow(common_synchronous.CommonSynchronousWorkflow):
     """Handles the workflow for the synchronous messaging pattern."""
 
-    async def handle_outbound_message(self, message_id: str, correlation_id: str, interaction_details: dict,
-                                      payload: str) -> Tuple[int, str]:
-        raise NotImplementedError()
+    def __init__(self,
+                 party_key: str = None,
+                 work_description_store: pa.PersistenceAdaptor = None,
+                 transmission: transmission_adaptor.TransmissionAdaptor = None,
+                 persistence_store_max_retries: int = None,
+                 routing: routing_reliability.RoutingAndReliability = None):
+        self.party_key = party_key
+        self.wd_store = work_description_store
+        self.transmission = transmission
+        self.persistence_store_retries = persistence_store_max_retries
+        super().__init__(routing)
+
+    async def handle_outbound_message(self,
+                                      from_asid: str,
+                                      message_id: str,
+                                      correlation_id: str,
+                                      interaction_details: dict,
+                                      payload: str,
+                                      work_description_object: Optional[wd.WorkDescription]) \
+            -> Tuple[int, str, Optional[wd.WorkDescription]]:
+        logger.info('001', 'Entered sync workflow for outbound message')
+        wdo = wd.create_new_work_description(self.wd_store,
+                                             message_id,
+                                             workflow.SYNC,
+                                             wd.MessageStatus.OUTBOUND_MESSAGE_RECEIVED
+                                             )
+        await wdo.publish()
+        if not from_asid:
+            return 400, '`from_asid` header field required for sync messages', None
+
+        try:
+            url, to_asid = await self._lookup_to_asid_details(interaction_details)
+        except Exception:
+            logger.error('009', 'Failed to retrieve details from spine route lookup')
+            await wdo.set_outbound_status(wd.MessageStatus.OUTBOUND_MESSAGE_PREPARATION_FAILED)
```

Comment:
The async express workflow uses `OUTBOUND_MESSAGE_TRANSMISSION_FAILED` when it fails to look up endpoint details. We should be consistent here.